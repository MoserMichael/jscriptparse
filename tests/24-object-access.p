

def makeComplex(re, im) {
    t = { 
             "re": re, 
             "im": im, 
             "add": def(c) { 
                return makeComplex(t.re + c.re, t.im + c.im)
              },
              "show": def() {
                println("re: {t.re} im: {t.im}")
              }
        }
    return t
}    

a=makeComplex(1,2)
a.show()

b=makeComplex(2,3)
b.show()

c=a.add(b)
c.show()

d=c.add(c)

d.show()


