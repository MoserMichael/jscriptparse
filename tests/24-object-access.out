re: 1 im: 2
re: 2 im: 3
re: 3 im: 5
re: 6 im: 10
